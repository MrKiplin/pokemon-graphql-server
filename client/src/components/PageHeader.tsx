import React from "react";
import { Header } from "semantic-ui-react";

interface PageHeaderProps {
  header: string;
  subHeader: string;
}

const PageHeader: React.FC<PageHeaderProps> = ({ header, subHeader }) => (
  <Header as="h1" textAlign="center">
    {header}
    <Header.Subheader>{subHeader}</Header.Subheader>
  </Header>
);

export default PageHeader;
