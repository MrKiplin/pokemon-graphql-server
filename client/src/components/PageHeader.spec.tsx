import { mount } from "enzyme";
import PageHeader from "./PageHeader";

describe("PageHeader", () => {
  it("Should match props and snapshot", () => {
    const component = mount(
      <PageHeader
        {...{ header: "test-header", subHeader: "test-sub-header" }}
      />
    );

    expect(component).toMatchSnapshot();
    expect(component.find("PageHeader").prop("header")).toEqual("test-header");
    expect(component.find("PageHeader").prop("subHeader")).toEqual(
      "test-sub-header"
    );
  });
});
