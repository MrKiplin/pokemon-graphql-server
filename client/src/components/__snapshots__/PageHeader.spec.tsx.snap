// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageHeader Should match props and snapshot 1`] = `ReactWrapper {}`;
